### DISCUSSION TITLE
The DW1000 is not responding, the interrupt pin will not trigger anymore

### DISCUSSION THREAD

**Discussion by lalalalang999** [bitcraze/discussions#805] [Q&A]
After running for a while, occasionally the DW1000 does not respond and the interrupt pin will no longer trigger. tag is stuck in the uwbTask task, please ask how to handle this exception. I don't have any troubleshooting ideas, would this be a case of DW1000 dying?

**Comment by ataffanel** [comment]
What version of the firmware are you running and have you made any modification to it? I have seen this kind of behavior before when the state machine in the Crazyflie gets out of sync with the DW1000 or when flags are not cleared properly.

**Reply by lalalalang999** [reply]
The version I'm running is the September 2019 Crazyflie firmware, I didn't make changes, but I took the UWB related code out and used it separately. What are the special circumstances that can cause this behavior to occur. Because the probability of this recurrence is very small, I read the DW1000 and state machine states when recurring, where the DW1000 state has bit 2 CPLOCK set to 1, bit 23 SLP2INIT set to 1, and bit HPDWARN set to 1. dwm->deviceMode is RX_MODE,wait4resp is How to handle this dead situation?

**Reply by ataffanel** [reply]
It looks like you are hitting this bug that was closed in August 2020: https://github.com/bitcraze/crazyflie-firmware/issues/368. I suggest you update your firmware to the latest version. The ticket of the bug contains some explanation about what was happening.

**Reply by lalalalang999** [reply]
Yes, it looks as if this is the cause of the problem. However, I have read the discussion about this problem and the final problem is that there is a wrong setting in uwbTask in reading the IRQ pin of DW1000, which leads to the inability to process the state, while I found this problem before, my own configuration about IRQ pin reading is no problem, so the problem that causes me is not the IRQ pin.
However, I have one more question, why here the binary semaphore changed to the task notification amount, if not change whether it will still lead to run stuck. The description of this point in the discussion, I do not understand.

**Reply by ataffanel** [reply]
I think the main fix was the GPIO fix, the notification instead of semaphore was likely just a try to see if using task notifications would help. I am not sure it does change anything actually.

Have you tried the latest firmware to see if you still have the bug there?

**Reply by lalalalang999** [reply]
I didn't solve the problem after replacing the task notification, so the bug didn't arise here. I am not using the latest firmware because I used the TDOA2 algorithm and I found that the 2023 and 2019 TDOA2 algorithms have changed significantly in the code.
But I saw an update about a bug on the dw1000 library operation, I suspect it might be caused here, I'm testing it and I'll let you know if I have results. Since I am using dw1000 channel 3, the library is wrong here in the 2020 register.    i'm not sure if this is the cause here.  [this
](https://github.com/bitcraze/libdw1000/blob/a77610327fb875a69543f3a8d19507683b68aff1/src/libdw1000.c#L1058C7-L1058C54)

**Reply by ataffanel** [reply]
I would advise you to try the latest firmware: we have mostly refactored the code when it comes to TDoA2 over the years, and it should still work pretty much the same way as it did back then. At least testing the latest firmware would allow us to know if there is still a bug left or if this has actually been fixed.
